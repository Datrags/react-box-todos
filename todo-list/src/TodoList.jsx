import {useState} from 'react'
import Todo from './Todo'
import NewTodoForm from './NewTodoForm'
const TodoList = () => {
    
    const [todos, setTodos] = useState([]);
    const createTodo = todo => {
        setTodos(todos => [...todos, todo]);
        console.log(todos);
    }

    const removeTodo = message => {
        setTodos(todos => todos.filter(todo => todo != message))
    }

    const TodoComponents = todos.map( (todo, index) => (
        <Todo key={index+"todo"} todo={todo} removeTodo={removeTodo}/>
    )) 

    return (
        <div>
            <NewTodoForm createTodo={createTodo}/>
            {TodoComponents}
        </div>)
}

export default TodoList;