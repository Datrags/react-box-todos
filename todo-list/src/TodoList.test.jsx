import React from "react";
import { render, fireEvent} from "@testing-library/react";
import "@testing-library/jest-dom"
import TodoList from "./TodoList";

it("should render without crashing", () => {
    render(<TodoList></TodoList>);
});

it("should match snapshot", () => {
    const {asFragment} = render(<TodoList/>);
    expect(asFragment()).toMatchSnapshot();
});
it("should add a todo", () => {
    const todoList = render(<TodoList/>);
    fireEvent.click(todoList.getByText("Submit"));
    let remBtn = todoList.getByText("x");
    expect(remBtn).toBeInTheDocument();
})

it("should have a working delete feature", () => {
    const todoList = render(<TodoList/>);
    fireEvent.click(todoList.getByText("Submit"));
    let remBtn = todoList.getByText("x");
    expect(remBtn).toBeInTheDocument();
    fireEvent.click(remBtn)
    expect(remBtn).not.toBeInTheDocument();
})