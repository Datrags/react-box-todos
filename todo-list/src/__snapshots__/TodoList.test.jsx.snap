// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="todo"
      >
        New Todo
      </label>
      <input
        id="todo"
        name="todo"
        placeholder="Enter New Todo"
        type="text"
        value=""
      />
      <button>
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
